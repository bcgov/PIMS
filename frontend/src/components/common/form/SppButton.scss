@import '../../../styles.scss';

.sppButton {
  .btn {
    height: 37px;
    width: 37px;
  }
}
.sppTooltip {
  pointer-events: all;
  .tooltip-inner {
    background-color: white;
    color: black;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    padding: 15px;
    text-align: left;
    .row {
      padding: 5px;
      &:hover {
        background-color: $accent-color;
      }
      margin-bottom: 10px;
      cursor: pointer;
      .marker {
        img {
          height: 32px;
        }
      }
      .label {
        padding-left: 0;
        display: flex;
        align-items: center;
      }
    }
  }
  .arrow::before {
    border-bottom-color: white !important;
    box-shadow: 2px grey;
  }
}
