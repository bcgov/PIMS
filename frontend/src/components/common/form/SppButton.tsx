import './SppButton.scss';

import React, { useState, useRef } from 'react';
import { ReactComponent as SppIcon } from 'assets/images/SPP.svg';
import { Col, Overlay, Tooltip, Button, Row, Image } from 'react-bootstrap';
import ClickAwayListener from 'react-click-away-listener';
import TooltipWrapper from '../TooltipWrapper';

interface ISppButtonProps {
  handleErpClick: () => void;
  handleSppClick: () => void;
  inEnhancedReferralProcess?: boolean;
  inSurplusPropertyProgram?: boolean;
}

const SppButton: React.FC<ISppButtonProps> = ({
  handleErpClick,
  handleSppClick,
  inEnhancedReferralProcess,
  inSurplusPropertyProgram,
}) => {
  const [visible, setVisible] = useState<boolean>(false);
  const target = useRef(null);

  const applyFilter = (callback: () => void) => {
    setVisible(false);
    callback();
  };

  const iconColor = inEnhancedReferralProcess
    ? '#cf3e4f'
    : inSurplusPropertyProgram
    ? '#a758cb'
    : '#3b99fc';

  return (
    <ClickAwayListener onClickAway={() => setVisible(false)}>
      <div className="sppButton">
        <TooltipWrapper toolTipId="map-filter-spp-tooltip" toolTip="Filter SPP/ERP Properties">
          <Button
            ref={target}
            className="close"
            onClick={() => setVisible(true)}
            style={{ backgroundColor: iconColor }}
          >
            <SppIcon title="spp" />
          </Button>
        </TooltipWrapper>
        <Overlay target={target.current!} show={visible} placement="bottom">
          {(props: any) => {
            return (
              <Tooltip id="overlay-spp" {...props} show={`${visible}`} className="sppTooltip">
                <h6>Properties On Surplus Properties Program</h6>
                <Row onClick={() => applyFilter(handleErpClick)}>
                  <Col xs={2} className="marker">
                    <Image src={require('assets/images/pins/marker-icon-2x-red.png')} />
                  </Col>
                  <Col className="label">Enhanced Referral Process</Col>
                </Row>
                <Row onClick={() => applyFilter(handleSppClick)}>
                  <Col xs={2} className="marker">
                    <Image src={require('assets/images/pins/marker-icon-2x-violet.png')} />
                  </Col>
                  <Col className="label">Surplus Properties List</Col>
                </Row>
              </Tooltip>
            );
          }}
        </Overlay>
      </div>
    </ClickAwayListener>
  );
};

export default SppButton;
