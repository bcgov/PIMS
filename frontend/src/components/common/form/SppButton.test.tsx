import React from 'react';
import renderer from 'react-test-renderer';
import { noop } from 'lodash';
import SppButton from './SppButton';

describe('SPP Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SppButton handleErpClick={noop} handleSppClick={noop} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
