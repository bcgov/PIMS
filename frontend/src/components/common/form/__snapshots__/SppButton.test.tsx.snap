// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SPP Button renders correctly 1`] = `
<div>
  <div
    className="sppButton"
  >
    <button
      className="close btn btn-primary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "backgroundColor": "#3b99fc",
        }
      }
      type="button"
    >
      <svg
        title="spp"
      >
        SPP.svg
      </svg>
    </button>
  </div>
</div>
`;
